pkg/services/profiles/add.go

```diff
@@ -0,0 +1,175 @@
+		return fmt.Errorf("failed to discover HelmRepository's name and namespace")
+	}
+
+	newRelease := helm.MakeHelmRelease(availableProfile, opts.Cluster, opts.Namespace)
+
+	files, err := gitProvider.GetRepoFiles(ctx, configRepoUrl, git.GetSystemPath(opts.Cluster), defaultBranch)
+	if err != nil {
+		return fmt.Errorf("failed to get files in '%s' for config repository '%s': %s", git.GetSystemPath(opts.Cluster), configRepoUrl.String(), err)
+	}
+
+	file, err := MakeManifestFile(files, newRelease, git.GetProfilesPath(opts.Cluster))
+	if err != nil {
+		return err
+	}
+
+	pr, err := gitProvider.CreatePullRequest(ctx, configRepoUrl, gitproviders.PullRequestInfo{
+		Title:         fmt.Sprintf("GitOps add %s", opts.Name),
+		Description:   fmt.Sprintf("Add manifest for %s profile", opts.Name),
+		CommitMessage: AddCommitMessage,
+		TargetBranch:  defaultBranch,
+		NewBranch:     automation.GetRandomName("wego-"),
+		Files:         file,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create the pull request: %s", err)
+	}
+	s.Logger.Actionf("Pull Request created: %s", pr.Get().WebURL)
+
+	if opts.AutoMerge {
+		s.Logger.Actionf("auto-merge=true; merging PR number %v", pr.Get().Number)
+		if err := gitProvider.MergePullRequest(ctx, configRepoUrl, pr.Get().Number, gitprovider.MergeMethodMerge, AddCommitMessage); err != nil {
+			return err
+		}
+	}
+
+	s.printAddSummary(opts)
+	return nil
+}
+
+func (s *ProfilesSvc) printAddSummary(opts AddOptions) {
+	s.Logger.Println("Adding profile:\n")
+	s.Logger.Println("Name: %s", opts.Name)
+	s.Logger.Println("Version: %s", opts.Version)
+	s.Logger.Println("Cluster: %s", opts.Cluster)
+	s.Logger.Println("Namespace: %s", opts.Namespace)
+	s.Logger.Println("")
+}
+
+func MakeManifestFile(files []*gitprovider.CommitFile, newRelease *v2beta1.HelmRelease, path string) ([]gitprovider.CommitFile, error) {
+	var content string
+	for _, f := range files {
+		if *f.Path == path {
+			if *f.Content == "" {
+				break
+			}
+
+			manifestByteSlice, err := splitYAML([]byte(*f.Content))
+			if err != nil {
+				return nil, err
+			}
+
+			for _, manifestBytes := range manifestByteSlice {
+				var r v2beta1.HelmRelease
+				if err := yaml.Unmarshal(manifestBytes, &r); err != nil {
+					return nil, err
+				}
+				if profileIsInstalled(r, *newRelease) {
+					return nil, fmt.Errorf("version %s of profile '%s' already exists in namespace %s", r.Spec.Chart.Spec.Version, r.Name, r.Namespace)
+				}
+			}
+			content = *f.Content + "\n---\n"
+		}
+	}
+	helmReleaseManifest, err := yaml.Marshal(newRelease)
+	if err != nil {
+		return nil, err
+	}
+	content = content + string(helmReleaseManifest)
+
+	return []gitprovider.CommitFile{
+		{
+			Path:    &path,
+			Content: &content,
+		},
+	}, nil
+}
+
+// splitYAML splits a manifest file that contains multiple YAML resources separated by '---'.
+func splitYAML(resources []byte) ([][]byte, error) {
```

Comment:
This code also ensures that each of the elements that were split are valid YAML which might not be obvious to a reader.